src/modules/push-notifications/push.notifications.service.ts

```diff
@@ -0,0 +1,240 @@
+            `accounts-000001_${epoch}`,
+            'address',
+            query,
+            async (items: AccountType[]) => {
+                const addresses = items.map(
+                    (item: AccountType) => item.address,
+                );
+                allAddresses.push(...addresses);
+            },
+        );
+
+        return allAddresses.map((address) => ({
+            address,
+            notificationSent: false,
+        }));
+    }
+
+    private async sendBatchNotifications(
+        addresses: string[],
+    ): Promise<boolean> {
+        const chainId = Number(this.apiConfigService.getChainId());
+        const notificationsApiUrl =
+            this.apiConfigService.getNotificationsApiUrl();
+        const notificationsApiKey =
+            this.apiConfigService.getNotificationsApiKey();
+
+        const payload: NotificationPayload = {
+            addresses,
+            chainId,
+            title: 'Energy Update',
+            body: 'You can now claim your rewards',
+            route: '/portfolio',
+            iconUrl:
+                'https://xexchange.com/assets/imgs/mx-logos/xexchange-logo@2x.webp',
+        };
+
+        const response = await axios.post(notificationsApiUrl, payload, {
+            headers: {
+                'Content-Type': 'application/json',
+                'x-notifications-api-key': notificationsApiKey,
+            },
+        });
+
+        if (response.status !== 201) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private async processNotificationBatch(
+        addresses: string[],
+        maxRetries: number,
+    ): Promise<{ successful: string[]; failed: string[] }> {
+        const batchSize =
+            await this.apiConfigService.getNotificationsBatchSize();
+
+        const successful: string[] = [];
+        let remainingAddresses = [...addresses];
+
+        for (
+            let retryCount = 0;
+            retryCount < maxRetries && remainingAddresses.length > 0;
+            retryCount++
+        ) {
+            if (retryCount > 0) {
+                this.logger.log(
+                    `Retry attempt ${retryCount} for ${remainingAddresses.length} addresses`,
+                );
+                await new Promise((resolve) => setTimeout(resolve, 10000));
+            }
+
+            const addressesToRetry = [...remainingAddresses];
+            remainingAddresses = [];
+
+            for (let i = 0; i < addressesToRetry.length; i += batchSize) {
+                const batch = addressesToRetry.slice(i, i + batchSize);
+                const success = await this.sendBatchNotifications(batch);
+
+                if (success) {
+                    successful.push(...batch);
+                } else {
+                    remainingAddresses.push(...batch);
+                }
+            }
+        }
+
+        return { successful, failed: remainingAddresses };
+    }
+
+    @GetOrSetCache({
```

Comment:
I don't think the caching is needed here; moreover it will expire by the time next cron is executed